src/app/views/settings/domain/domain.controller.coffee

```diff
@@ -1,31 +1,68 @@
-@App.controller 'SettingsDomainController', (toastr, MnoeTenant) ->
+@App.controller 'SettingsDomainController', ($filter, $scope, $window, toastr, MnoConfirm, MnoeTenant) ->
   'ngInject'
   vm = this
   vm.tenant = {}
+  vm.originalTenant = angular.toJson(vm.tenant)
   vm.ssl = {}
+  vm.originalSsl = angular.toJson({})
   vm.isLoading = true
 
+  # has any of the form changed?
+  changedForm = () ->
+    (vm.originalTenant != angular.toJson(vm.tenant)) || (vm.originalSsl != angular.toJson(vm.ssl))
```

Comment:
I think it would be better to use https://docs.angularjs.org/api/ng/function/angular.equals